ledger-tool/src/main.rs

```diff
@@ -1112,6 +1120,76 @@ fn main() {
                 }
             }
         }
+        ("capitalization", Some(arg_matches)) => {
+            let dev_halt_at_slot = value_t!(arg_matches, "halt_at_slot", Slot).ok();
+            let process_options = ProcessOptions {
+                dev_halt_at_slot,
+                new_hard_forks: hardforks_of(arg_matches, "hard_forks"),
+                poh_verify: false,
+                ..ProcessOptions::default()
+            };
+            let genesis_config = open_genesis_config_by(&ledger_path, arg_matches);
+            match load_bank_forks(arg_matches, &ledger_path, &genesis_config, process_options) {
+                Ok((bank_forks, _leader_schedule_cache, _snapshot_hash)) => {
+                    let slot = bank_forks.working_bank().slot();
+                    let bank = bank_forks.get(slot).unwrap_or_else(|| {
+                        eprintln!("Error: Slot {} is not available", slot);
+                        exit(1);
+                    });
+
+                    use solana_sdk::native_token::LAMPORTS_PER_SOL;
+                    use std::fmt::{Display, Formatter, Result};
+                    pub struct Sol(u64);
+
+                    impl Display for Sol {
+                        fn fmt(&self, f: &mut Formatter) -> Result {
+                            write!(
+                                f,
+                                "{}.{:09} SOL",
+                                self.0 / LAMPORTS_PER_SOL,
+                                self.0 % LAMPORTS_PER_SOL
+                            )
+                        }
+                    }
+
+                    let computed_capitalization: u64 = bank
+                        .get_program_accounts(None)
+                        .into_iter()
+                        .filter_map(|(_pubkey, account)| {
+                            if account.lamports == u64::max_value() {
+                                return None;
+                            }
+
+                            let is_specially_retained =
+                                solana_sdk::native_loader::check_id(&account.owner)
```

Comment:
@jackcmay I'm not that familiar with loaders... Well, loaders can have similar problem as this pr? What happens to the balance if one of loaders is updated? (or not updated ever, first of all?)

Reply:
Well, it seems that I can't change of `account.lamports` first of all:

```
$ solana --url http://localhost:8899 pay -v Stake11111111111111111111111111111111111111 0.82643
RPC URL: http://localhost:8899
Default Signer Path: /home/ryoqun/.config/solana/id.json
Error: TransactionError::InstructionError(0, ExecutableLamportChange)
```

Reply:
Once marked executable you can't change its lamports

Reply:
And, iirc the `executable` bit can't be off after that, too? :)

Reply:
Correct